app/scripts/lib/channels/notifier.js

```diff
@@ -57,6 +93,13 @@ define(function (require, exports, module) {
     },
 
     triggerRemote: function (event, data) {
+      if (SCHEMATA[event]) {
+        if (! Validate.isDataValid(data, SCHEMATA[event])) {
+          throw new Error('Invalid data for event ' + event);
+        }
+      } else if (data !== null && data !== undefined) {
```

Comment:
Can you use the underscore `_.isNull` and `_.isUndefined` here? @vladikoff keeps hitting me over the head for not using them, and I'm passing on the knocks to keep consistency.